File: 216-combination-sum-iii/combination-sum-iii.js
var combinationSum3 = function(k, n) {
    const result = [];

    const backtrack = (start, target, currentCombination) => {
        if (target === 0 && currentCombination.length === k) {
            result.push([...currentCombination]);
            return;
        }

        for (let i = start; i <= 9; i++) {
            if (target - i >= 0) {
                currentCombination.push(i);
                backtrack(i + 1, target - i, currentCombination);
                currentCombination.pop();
            }
        }
    };

    backtrack(1, n, []);

    return result;
};

// Example usage:
const k = 3;
const n = 7;
const result = combinationSum3(k, n);
console.log(result);

File: 216-combination-sum-iii/combination-sum-iii.cpp
class Solution {
public:
  vector<vector<int>>res;

   void backtrack(int k, int n, int start, vector<int>path){
       if(n==0 && path.size() == k){
        res.push_back(path);
        return;
       }
           if (n < 0 || path.size() > k) return; 
       for(int i=start;i<=9;i++){
        path.push_back(i);
        backtrack(k,n-i,i+1,path);
        path.pop_back();
       }
   }


    vector<vector<int>> combinationSum3(int k, int n) {
        vector<int> path;
        backtrack(k, n, 1, path);
        return res;
    }
};